packages/editor/src/browser/navigation/navigation-location-service.ts

```diff
@@ -233,4 +285,64 @@ ${this.stack.map((location, i) => `${i}: ${JSON.stringify(NavigationLocation.toO
 ----- o -----`;
     }
 
+    /**
+     * Get the recently closed editors stack in chronological order.
+     *
+     * @returns readonly closed editors stack.
+     */
+    recentlyClosedEditors(): ReadonlyArray<RecentlyClosedEditor> {
+        return this.recentlyClosedEditorsStack;
+    }
+
+    /**
+     * Get the last recently closed editor.
+     *
+     * @returns the recently closed editor if it exists.
+     */
+    getLastClosedEditor(): RecentlyClosedEditor | undefined {
+        return this.recentlyClosedEditorsStack[this.recentlyClosedEditorsStack.length - 1];
+    }
+
+    /**
+     * Add the recently closed editor to the history.
+     *
+     * @param widget the recently closed editor.
+     */
+    addToRecentlyClosedEditors(widget: EditorWidget): void {
```

Comment:
As mentioned offline, we should only have a single public API for callers to add items into the stack.
(`pushRecentlyClosedEditors`).